services/localfile/server/shred_test.go

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2019 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+package server
+
+import (
+	"context"
+	pb "github.com/Snowflake-Labs/sansshell/services/localfile"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"os"
+	"testing"
+)
+import "github.com/Snowflake-Labs/sansshell/testing/testutil"
+
+func Test_Shred(t *testing.T) {
+	if shredIsUnavailable() {
+		t.Skipped()
+		return
+	}
+
+	ctx := context.Background()
+	conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	testutil.FatalOnErr("grpc.DialContext(bufnet)", err, t)
+
+	t.Cleanup(func() { conn.Close() })
+	client := pb.NewLocalFileClient(conn)
+
+	data := "data"
+	fileName := createTempFile(data, t)
+
+	_, err = client.Shred(ctx, &pb.ShredRequest{
+		Filename: fileName,
+		Force:    false,
+		Zero:     false,
+		Remove:   false,
+	})
+	testutil.FatalOnErr("shred", err, t)
+
+	if getFileContent(t, fileName) == data {
```

Comment:
nit: We should fail the test if the shredded file contains any data, not necessarily the original data.

Reply:
Thanks, I changed this to include zero-ing pass and checking that all bytes are zeros after. As a side note Shred will actually extend the file size to `((size_of_original_data / block_size) + 1) * block_size`, typically block size is 512/4096 bytes on HDDs (4k on modern ones), so for a <block_size file it will be extended to 4kb (filled with zeros/random data depending on Shred's settings).